### ISSUE TITLE
Node attributes

### ISSUE THREAD

**Issue by LIANGKE23** [kkteru/grail#2]
Is there any way to add attributes inside of your model?

**Comment by egdenis** [comment]
Yes, currently the default is a binary flag to use node embeddings as node features but you can easily modify it to use external attributes. You can pass in your node attributes using the [n_feats flag](https://github.com/kkteru/grail/blob/7180a5208e33e69226e7f5eebee220999d0a959e/subgraph_extraction/datasets.py#L156) of the _prepare_features_new function. Let me know if you have any more questions

**Comment by LIANGKE23** [comment]
Ok! I will try it today.
I have another question for testing.
Currently, it seems that the testset is the groudtruth we want to predict. And the results will give me the evaluation metric value such as AUC, MRR, Hit@x.
Is there any way to modify the model as the prediction model?
For example, currently we need the testset be in the format of (x,r,y), and we doing the head prediction using your model, and it will tell us that x is in the kth of the head prediction results. But can we modify it to the model which only takes (r,y) as input, and output the x we want?

**Comment by LIANGKE23** [comment]
And for the node features, for example, in my graph, there are 6 attributes for the nodes, what is the format of the attribues file I should build? A mat? or npy? with the rows for attributes? or?
Can you give an example?

**Comment by LIANGKE23** [comment]
> 
> 
> Yes, currently the default is a binary flag to use node embeddings as node features but you can easily modify it to use external attributes. You can pass in your node attributes using the [n_feats flag](https://github.com/kkteru/grail/blob/7180a5208e33e69226e7f5eebee220999d0a959e/subgraph_extraction/datasets.py#L156) of the _prepare_features_new function. Let me know if you have any more questions

And for the node features, for example, in my graph, there are 6 attributes for the nodes, what is the format of the attribues file I should build? A mat? or npy? with the rows for attributes? or?
Can you give an example?
